gson/src/main/java/com/google/gson/stream/JsonReader.java

```diff
@@ -406,24 +394,29 @@ public void endObject() throws IOException {
   }
 
   /**
-   * Returns true if the current array or object has another element.
+   * Returns peeked if not PEEKED_NONE, if PEEKED_NONE then doPeek() is returned instead
    */
-  public boolean hasNext() throws IOException {
+  private int peeked() {
     int p = peeked;
     if (p == PEEKED_NONE) {
-      p = doPeek();
+         p = doPeek();
```

Comment:
It looks like the indentation was changed here by accident:
```suggestion
      p = doPeek();
```